docs/Configuration.md

```diff
@@ -0,0 +1,119 @@
+# Configuring Just Saying
+
+## The publish subscribe pattern
+> In software architecture, publish–subscribe is a messaging pattern where senders of messages, called publishers, do not program the messages to be sent directly to specific receivers, called subscribers, but instead categorize published messages into classes without knowledge of which subscribers, if any, there may be. Similarly, subscribers express interest in one or more classes and only receive messages that are of interest, without knowledge of which publishers, if any, there are. [(wikipedia)](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern)
+
+# Configuring a single publisher and a single subscriber
+
+## Configuring the publisher
+
+- In this configuration the following AWS resources will be created
+    - a SNS topic of name `orderplacedevent`
+    - a SQS queue of name `orderplacedevent`
+    - a SNS topic subscription for queue `orderplacedevent` to topic `orderplacedevent`
+
+- In `startup.cs` add the following
+```cs
+    services.AddJustSaying(config =>
+    {
+        config.Publications(x =>
+        {
+            // Creates the SNS Topic
+            x.WithTopic<OrderPlacedEvent>();
+        });
+    }
+```
+
+## Configuring the subscriber
+
+- In `startup.cs` add the following
+```cs
+    services.AddJustSaying(config =>
+    {
+        config.Subscriptions(x =>
+        {
+            // creates the SQS queue
+            // creates the SQS to SNS subscription
+            x.ForTopic<OrderPlacedEvent>();
+        });
+    }
+    // Add a handler responsibile for handling the event
+    services
+        .AddJustSayingHandler<OrderPlacedEvent, OrderPlacedEventHandler>();
+
+    // Add a background service that is listening for messages related to the above subscriptions
+    services.AddHostedService<Subscriber>();
+```
+
+*Resulting infrastructure*
+
+![Single Publisher - Single Subscriber](SinglePublisher_SingleSubscriber.png)
```

Comment:
At a later point, we'd need the diagram generation documented so we know how to regenerate it easily.

Reply:
See [docs/Documentation_generation.md](docs/Documentation_generation.md)